library/src/main/kotlin/com/chuckerteam/chucker/internal/ui/transaction/TransactionPayloadFragment.kt

```diff
@@ -200,15 +235,56 @@ internal class TransactionPayloadFragment :
     override fun onQueryTextSubmit(query: String): Boolean = false
 
     override fun onQueryTextChange(newText: String): Boolean {
+        var listOfScrollableIndex = listOf<Int>()
         if (newText.isNotBlank() && newText.length > NUMBER_OF_IGNORED_SYMBOLS) {
-            payloadAdapter.highlightQueryWithColors(
+            listOfScrollableIndex = payloadAdapter.highlightQueryWithColors(
                 newText,
                 backgroundSpanColor,
                 foregroundSpanColor
             )
         } else {
             payloadAdapter.resetHighlight()
         }
+
+        scrollableIndices.clear()
+        when {
+            listOfScrollableIndex.isEmpty() -> {
+                // scroll to top
+                currentSearchScrollIndex = 0
+                payloadBinding.payloadRecyclerView.smoothScrollToPosition(0)
```

Comment:
I'm unsure this is good from the UX point of view (i.e. the user is searching and you keep on scrolling up and down).

Reply:
I prefer the implemented way, quickly jumping to the position. However, I can think of two solutions here. 

We can increase the initial delay.

Or we can only scroll after the IME_ACTION_DONE or keyboard closed - along with the summary of search results in the app bar.

Reply:
Sorry my comment was related to the re-scroll to top once `listOfScrollableIndex.isEmpty`

Reply:
Oh okay, yeah let's not scroll in that condition. Will do the change.